packages/mobile/src/screens/edit-track-screen/EditTrackScreen.tsx

```diff
@@ -4,86 +4,203 @@ import type { UploadTrack } from '@audius/common'
 import {
   creativeCommons,
   formatPrice,
+  isPremiumContentUSDCPurchaseGated,
   useUSDCPurchaseConfig
 } from '@audius/common'
 import { Formik } from 'formik'
-import * as Yup from 'yup'
+import { z } from 'zod'
+import { toFormikValidationSchema } from 'zod-formik-adapter'
 
 import { useRemoteVar } from 'app/hooks/useRemoteConfig'
 
 import { EditTrackNavigator } from './EditTrackNavigator'
+import { TRACK_PREVIEW } from './fields/AccessAndSaleField/PremiumRadioField/TrackPreviewField'
+import { TRACK_PRICE } from './fields/AccessAndSaleField/PremiumRadioField/TrackPriceField'
 import type { FormValues, EditTrackScreenProps } from './types'
 const { computeLicenseVariables, ALL_RIGHTS_RESERVED_TYPE } = creativeCommons
 
+const errorMessages = {
+  title: 'Your track must have a name',
+  genre: 'Genre is required',
+  artwork: 'Artwork is required',
+  description: 'Description cannot exceed 1000 characters',
+  previewStartThirtyBeforeEnd:
+    'Preview must start at least 30 seconds before the end of the track.',
+  previewStartZero:
+    'Preview must start at 0 since the track is less than 30 seconds.'
+}
+
 const useEditTrackSchema = () => {
   const { minContentPriceCents, maxContentPriceCents } =
     useUSDCPurchaseConfig(useRemoteVar)
   return useMemo(
+    /**
```

Comment:
this does feel fairly antithetical like youre saying. im okay rolling forward with this temporarily so we can have a successful launch, but adding an ticket to look deeper into this issue: https://linear.app/audius/issue/C-3292/review-mobile-zod-schema